app/controllers/Auth.scala

```diff
@@ -74,14 +74,15 @@ final class Auth(
   def loginLang = LangPage(routes.Auth.login)(serveLogin)
 
   private def serveLogin(using ctx: Context) = NoBot:
-    val referrer = get("referrer") flatMap env.api.referrerRedirect.valid
-    val switch   = get("switch")
-    referrer ifTrue ctx.isAuth ifTrue switch.isEmpty match
-      case Some(url) => Redirect(url) // redirect immediately if already logged in
-      case None =>
-        val prefillUsername = lila.security.UserStrOrEmail(~switch.filter(_ != "1"))
-        val form            = api.loginFormFilled(prefillUsername)
-        Ok.page(html.auth.login(form, referrer)).map(_.withCanonical(routes.Auth.login))
+    RedirectToProfileIfLoggedIn:
+      val referrer = get("referrer") flatMap env.api.referrerRedirect.valid
+      val switch   = get("switch")
+      referrer ifTrue ctx.isAuth ifTrue switch.isEmpty match
+        case Some(url) => Redirect(url) // redirect immediately if already logged in
```

Comment:
This probably break the case where you had a referrer. I don’t think this change here is necessary

Reply:
I didnt change anything in this line. Its green because of indentation change. I just added RedirectToProfileIfLoggedIn: line. And the referrer works fine

Reply:
https://github.com/lichess-org/lila/assets/95964955/fd4971b1-7a72-4534-94e9-1f89d5e71854